jupyter/d2l-java/chapter_mlp/mlp.ipynb

```diff
@@ -0,0 +1,1006 @@
+       "y: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0],\n",
+       "mode: 'lines',\n",
+       "type: 'scatter',\n",
+       "name: '',\n",
+       "};\n",
+       "\n",
+       "\n",
+       "var data = [ trace0];\n",
+       "Plotly.newPlot(target_f6f5dd7bfbeb4b35a5d625ad07f33905, data, layout);\n",
+       "})</script>\n"
+      ],
+      "text/plain": [
+       "tech.tablesaw.plotly.components.Figure@45d76670"
+      ]
+     },
+     "execution_count": 74,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "try(GradientCollector collector = MxEngine.getInstance().newGradientCollector()){\n",
+    "    y = Activation.relu(x);\n",
+    "    collector.backward(y); \n",
+    "}\n",
+    "\n",
+    "NDArray res = x.getGradient();\n",
+    "\n",
+    "int xLength = (int) x.size();\n",
+    "int yLength = (int) res.size();\n",
+    "\n",
+    "float[] X = new float[xLength];\n",
+    "float[] Y = new float[yLength];\n",
+    "\n",
+    "X = x.toFloatArray();\n",
+    "Y = res.toFloatArray();\n",
+    "\n",
+    "Table data = Table.create(\"Data\").addColumns(FloatColumn.create(\"X\", X), FloatColumn.create(\"grad of relu\", Y));\n",
```

Comment:
You should split this across multiple lines for better readability:
```
Table.create("Data")
    .addColumns(
        FloatColumn.create("X", x.toFloatArray())
        FloatColumn.create("relu(X)", y.toFloatArray())
    );
```

You can also clean this up further by just combining the toFloatArray call into the column creation like I did because it doesn't take up too much space. Then, you save a few extra lines of code since you don't need xLength or X any more.

These suggestions also apply to the other examples on the page